src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -608,6 +608,8 @@ private Task ProcessSettingsFrameAsync()
 
                 _clientSettings.ParseFrame(_incomingFrame);
 
+                _frameWriter.WriteSettingsAckAsync(); // Ack before we update the windows, they could send data immediately.
```

Comment:
do we want this as fire and forget? Why not await it?

Reply:
I'm on the fence on this. On one hand, there are several places already that we write small frames without awaiting them, and I think that's fine. On the other hand, other methods called by ProcessFrameAsync do return framewriter task, so it's more consistent to continue to do so.

I don't like introducing a new async state machine, but maybe that won't be to big of an issue in practice if settings frames are uncommon enough. Another alternative is just storing the Task returned by WriteSettingsAckAsync in a local and returning it later.

Reply:
Storing it and returning it later seems like a good compromise.